spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java

```diff
@@ -127,6 +131,23 @@ private void executeIndexOp(
                 flintIndexMetadataService);
         flintIndexOpAlter.apply(indexMetadata);
         break;
+      case VACUUM:
+        // Try to perform drop operation first
+        FlintIndexOp tryDropOp =
+            new FlintIndexOpDrop(
+                stateStore, dispatchQueryRequest.getDatasource(), emrServerlessClient);
+        try {
+          tryDropOp.apply(indexMetadata);
+        } catch (IllegalStateException e) {
```

Comment:
Do we want to ignore only IllegalStateException or any kind of error?

Reply:
I think only the IllegalStateException thrown from validation? If any exception occurred during operation and commit, there is no need to continue.